core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -875,6 +877,20 @@ public final Sql sql(String sql) {
     sql(sql).ok();
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-4092">[CALCITE-4092]
+   * NPE using WITH clause without a corresponding SELECT FROM</a>. */
+  @Test void testWithNotSelected() {
+    final String sql = "with emp2 as (select max(empno) as empno from emp where empno > 10)\n"
+        + "select * from emp where empno < emp2.empno";
+    try {
+      sql(sql).ok();
```

Comment:
This test probably belongs more to `SqlValidatorTest` which has support to test for error (and validate error messages).`

Reply:
You're right. I moved the test to SqlValidatorTest and made some adjustments accordingly.